backend/src/routes/api/notebooks/index.ts

```diff
@@ -61,39 +63,78 @@ module.exports = async (fastify: KubeFastifyInstance) => {
   );
 
   fastify.post(
-    '/:namespace',
+    '/',
     secureRoute(fastify)(
       async (
         request: FastifyRequest<{
-          Params: {
-            namespace: string;
-          };
-          Body: Notebook;
+          Body: NotebookData;
         }>,
+        reply: FastifyReply,
       ) => {
-        return createNotebook(fastify, request);
+        const notebookData = request.body;
+        const username = await getUserName(fastify, request);
+
+        if (notebookData.state !== NotebookAction.Create) {
+          reply.status(400).send('Failed to create the Notebook');
+        }
+
+        return createNotebook(fastify, username, notebookData).catch((e) => {
+          fastify.log.error(
+            `Failed to create notebook, ${e.response?.data?.message || e.message}}`,
+          );
+          reply.status(400).send(e.response?.data?.message || e.message);
+        });
+      },
+    ),
+  );
+
+  fastify.put(
```

Comment:
>    * `PUT` "Restart" Notebook (which takes form data instead of a CR) -- honestly this could probably be part of the `POST` call and just check if we have a notebook, update/restart it, if we don't create it

Now that @andrewballantyne has reminded me that starting the notebook takes the same data as the create, I think he's right.  Can we just get rid of the PUT altogether and do a get at the beginning of the post and act accordingly?

Reply:
sounds right to me, having that kind of logic in the post seems right, I would agree, what do you think @andrewballantyne ?

Reply:
Yup, that seems the most logical... it'll be doing a lot of the same code for create or update & start... so it definitely seems wise to make 1 endpoint to avoid having to figure out how to share 90% of the code between the two flows.